tests/test_group_by.cc

```diff
@@ -383,7 +385,7 @@ double mean(const std::vector<double> &xs) {
   return mean / cast::to_double(xs.size());
 }
 
-long int number_of_digits(double x) { return lround(ceil(log10(x))); }
+long int number_of_digits(double x) { return (long int)floorl(log10(x) + 1.); }
```

Comment:
I see, this was fixing the edge cases were `log(x)` falls on an integer? Nice catch. We should change the casting to `static_cast<long int>(...)` though, or you could stick with the `lround` approach,
```suggestion
long int number_of_digits(double x) { return lround(floor(log10(x))); }
```

Reply:
lround rounds up or down, but it should be down only. So I'll implement the casting as 
`long int number_of_digits(double x) {static_cast<long int>(lround(ceil(log10(x))));}`